src/main/java/it/gov/pagopa/pu/fileshare/service/FileStorerService.java

```diff
@@ -0,0 +1,78 @@
+package it.gov.pagopa.pu.fileshare.service;
+
+import it.gov.pagopa.pu.fileshare.config.FoldersPathsConfig;
+import it.gov.pagopa.pu.fileshare.exception.custom.FileUploadException;
+import it.gov.pagopa.pu.fileshare.exception.custom.InvalidFileException;
+import it.gov.pagopa.pu.fileshare.util.AESUtils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Slf4j
+@Service
+public class FileStorerService {
+  private final FoldersPathsConfig foldersPathsConfig;
+  private final String fileEncryptPassword;
+
+  public FileStorerService(FoldersPathsConfig foldersPathsConfig,
+    @Value(("${app.fileEncryptPassword}")) String fileEncryptPassword) {
+    this.foldersPathsConfig = foldersPathsConfig;
+    this.fileEncryptPassword = fileEncryptPassword;
+  }
+
+  private Path getFilePath(String relativePath, String filename) {
+    String basePath = foldersPathsConfig.getShared()+relativePath;
```

Comment:
using the following instructions we could avoid to be aware that shared folder or relativePath have `/` between them
```suggestion
    Path basePath = Path.of(foldersPathsConfig.getShared(),relativePath);
```